src/encoder.rs

```diff
@@ -347,6 +349,51 @@ impl<T: Pixel> FrameState<T> {
     }
   }
 
+  pub fn compute_activity_mask(&mut self) {
+    //activity mask is computed only for luma plane
+
+    let luma_plane = &self.input.planes[0];
+    let PlaneConfig { width, height, .. } = luma_plane.cfg;
+    let luma = &luma_plane.data;
+    let PlaneConfig { xdec, ydec, .. } = self.activity_mask.cfg;
+    assert!(xdec == 3);
+    assert!(ydec == 3);
+    let width = (width >> xdec) << xdec;
+    let height = (height >> ydec) << ydec;
+    let mut mean = vec![0u32; (width >> xdec) * (height>>ydec)];
+    let mut var = vec![0u32; (width >> xdec) * (height>>ydec)];
+
+    for r in 0..height {
+      for c in 0..width {
+        let luma_offset = luma_plane.index(c, r);
+        let act_offset = (r>>ydec) * (width>>ydec) + (c>>xdec);
+        let temp_luma: u32 = luma[luma_offset].into(); 
+        mean[act_offset] += temp_luma;
+      }
+    }
+
+    mean.iter_mut().for_each(|sum| *sum >>= xdec+ydec);
+
+    for r in 0..height {
+      for c in 0..width {
+        let luma_offset = luma_plane.index(c, r);
+        let act_offset = (r>>ydec) * (width>>ydec) + (c>>xdec);
+        let temp_luma: u32 = luma[luma_offset].into();
+        let temp_mean: u32 = mean[act_offset];
+        var[act_offset] += (temp_luma as i32 - temp_mean as i32).pow(2) as u32;
+      }
+    }
+    var.iter_mut().for_each(|sum| *sum = (((*sum >> (xdec+ydec)) as f32).sqrt()) as u32 * 2);
+
+    let max = var.iter().max().unwrap();
+    let activity_mask: &mut [u16] = &mut *self.activity_mask.data;
+    activity_mask
+      .iter_mut()
+      .zip(var.iter())
+      .map(|(var1, var2)| *var1 = if *max!=0 {(1<<16 * var2 / max)} else {0} as u16 )
```

Comment:
Clippy is rightly warning to be wary of operator precedence near a shift. This line is subtly different from the intended calculation. The odd suggestion given by Clippy demonstrates how it is being parsed.
```suggestion
      .map(|(var1, var2)| *var1 = if *max!=0 {(1<<16) * var2 / max} else {0} as u16 )
```